pyannote/audio/tasks/joint_task/speaker_diarization_and_embedding.py

```diff
@@ -0,0 +1,1032 @@
+            to corresponding file to be drawn
+        rng : random.Random
+            Random number generator
+        duration: float
+            duration of the chunk to draw
+        """
+        # select one file at random (wiht probability proportional to its annotated duration)
+        file_id = np.random.choice(file_ids, p=prob_annotated_duration)
+        # find indices of annotated regions in this file
+        annotated_region_indices = np.where(
+            self.annotated_regions["file_id"] == file_id
+        )[0]
+
+        # turn annotated regions duration into a probability distribution
+        prob_annotaded_regions_duration = self.annotated_regions["duration"][
+            annotated_region_indices
+        ] / np.sum(self.annotated_regions["duration"][annotated_region_indices])
+
+        # seletect one annotated region at random (with probability proportional to its duration)
+        annotated_region_index = np.random.choice(annotated_region_indices,
+                                                  p=prob_annotaded_regions_duration
+                                                  )
+
+        # select one chunk at random in this annotated region
+        _, _, start, end = self.annotated_regions[annotated_region_index]
+        start_time = rng.uniform(start, end - duration)
+
+        return (file_id, start_time)
+
+    def draw_embedding_chunk(self, klass : Text,
+                               classes : List[Text],
+                               duration : float) -> tuple:
+        """Sample one chunk for the embedding task
+
+        Parameters
+        ----------
+        klass: Text
+            current class of speakers from which to draw a sample
+        classes: List[Text]
+            list of all the global speaker labels, in the same order than the list
+            defined in the task specification
+        duration: float
+            duration of the chunk to draw
+
+        Return
+        ------
+        tuple:
+            file_id:
+                the file id to which the sampled chunk belongs
+            start_time:
+                start time of the sampled chunk
+        """
+        # get index of the current class in the order of original class list
+        class_id = classes.index(klass)
+        # get segments for current class
+        class_segments_idx = self.annotations["global_label_idx"] == class_id
+        class_segments = self.annotations[class_segments_idx]
+
+        # sample one segment from all the class segments:
+        segments_duration = class_segments["end"] - class_segments["start"]
+        segments_total_duration = np.sum(segments_duration)
+        prob_segments = segments_duration / segments_total_duration
+        segment = np.random.choice(class_segments, p=prob_segments)
+
+        # sample chunk start time in order to intersect it with the sampled segment
+        start_time = np.random.uniform(segment["start"] - duration / 2, segment["start"])
```

Comment:
I would actually sample between 
* `segment["start"] - duration`, because the choice of 50% of `duration` is a bit arbitrary and might lead to a bias in the distribution of speech within a chunk. 
* and `segment["end"]`, because using `segment["start"]` would definitely lead to a similar bias as well.